src/ripple/resource/impl/Consumer.cpp

```diff
@@ -114,10 +114,15 @@ Consumer::warn()
 }
 
 bool
-Consumer::disconnect()
+Consumer::disconnect(beast::Journal const& j)
 {
     assert(m_entry != nullptr);
-    return m_logic->disconnect(*m_entry);
+    bool const d = m_logic->disconnect(*m_entry);
+    if (d)
+    {
+        JLOG(j.debug()) << "disconnecting " << entry().to_string();
```

Comment:
The rest of this function uses `*m_entry`, so using `entry()` here (which is equivalent other than asserting again) seems odd.